src/test/java/net/datafaker/providers/base/AbstractBasicProviderTest.java

```diff
@@ -0,0 +1,64 @@
+package net.datafaker.providers.base;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.function.Supplier;
+
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public abstract class AbstractBasicProviderTest<T extends BaseFaker> extends BaseFakerTest<BaseFaker> {
+    @SuppressWarnings("unchecked")
+    protected List<String> getBaseList(String key) {
+        return (List<String>) faker.fakeValuesService().fetchObject(key, faker.getContext());
+    }
+
+    @ParameterizedTest(name = "{0}")
+    @MethodSource("providerListTest")
+    protected void testProviderList(TestSpec testSpec) {
+        // Given
+        List<String> actual = getBaseList(testSpec.key);
+        // When
+        String item = (String) testSpec.supplier.get();
+        // Then
+        assertThat(item).isNotEmpty();
+        assertThat(actual).anyMatch(item::equals);
```

Comment:
SonarList suggests tweaking this similar to:
```java
        assertThat(item).as("Check item isn't empty").isNotEmpty();
        assertThat(actual).as("Check actual list contains the item").anyMatch(item::equals);
```

Do you think that helps?

https://github.com/joansmith/sonar-java/blob/master/java-checks/src/main/resources/org/sonar/l10n/java/rules/squid/S2698.html

Reply:
I'm not sure about having 
```java
        assertThat(item).isNotEmpty();
```
Will not the `anyMatch` condition anyway fail in case `actual` is empty?

Reply:
in fact assertj api is mostly self explanatory, so i don't think an extra description is required here...
i would say it's optional... so up to you

Reply:
Yes. I did a quick test setting actual to Collections.emptyList() and it does fail in that case.

Reply:
Well okay the output could be more specific.
```text
java.lang.AssertionError: [Check actual list contains the item] 
Expecting any elements of:
  []
to match given predicate but none did.
```

I could add another assertion that actual isn't empty.

Reply:
ok, makes sense

Reply:
Okay, this is addressed now too.

Reply:
I'm not a fan of using those `as` things, I only use them in very specific cases, I think the default error messages are pretty self-explanatory, but on the other hand, if you think it's nice to have them (I thing that issue, RSPEC-2698, is really not useful, and adds more maintenance burden than it's worth. But I'm not a fan of SonarQube/Linters/etc in general :-) )

Reply:
No need to remove them, I'm fine with it either way, it just feels a bit like duplication to me, that's all.